### PR TITLE
fix: also copy user labels + workflow-template. Fixes  #6129

### LINKED ISSUE #6129: Resubmit an archived workflow from UI does not keep labels
## Summary 

From an archived workflow, click on "Resubmit" in the UI

The created workflow do not have the appropriate labels

/!\ the bug happens only for archived workflows (when you resubmit a "normal" workflow, the labels are kept)

Be cautious, some labels should not be copied (like workflows.argoproj.io or workflows.argoproj.io/phase)

however, the labels placed manually by the user should be duplicated

rational : labels are used to search workflows, so if they do not have labels they do not appear in search results anymore ...

I am not sure to consider bug or enhancement ...

root cause : when you resubmit an archived workflow, the UI is different from a normal workflow (in archive, you use a YAML where the labels are not preloaded whereas in normal it is just asking a confirmation)

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by christopheblin**
fix #6129

Not sure this is the most elegant solution

I did test my solution only thanks to the chrome devtools on the runnning UI in 3.1.0-rc12

### CODE REVIEW

**Comment by christopheblin** [comment]
I do not understand why Semantic PR check is failing (the title do start with "fix ..." which should be enough)

**Comment by christopheblin** [comment]
now, in addition to the semantic PR check, the CI breaks with:

CI / e2e k8sapi

```
# Wait for workflow controller
until lsof -i :9090 > /dev/null ; do sleep 10s ; done
Error: The action has timed out.
```

CI / UI
```
Run git diff --exit-code
...
Error: Process completed with exit code 1.
```

what should I do for the 3 things ?

File: ui/src/app/archived-workflows/components/archived-workflow-details/archived-workflow-details.tsx [hunk 1]
```diff
@@ -241,6 +242,15 @@ export class ArchivedWorkflowDetails extends BasePage<RouteComponentProps<any>,
             </>
         );
     }
+                            
+    private cloneLabelsSafely() {
+        //we want to copy all labels except the ones that are added during workflow execution
+        const labelsArray = Object.keys(this.state.workflow.metadata.labels)
+            .map(k => ({ key: k, value: this.state.workflow.metadata.labels[k] }))
+            .filter(e => !e.key.startsWith("workflows.argoproj.io/") || e.key === "workflows.argoproj.io/workflow-template");
```
Comment (alexec) [inline ui/src/app/archived-workflows/components/archived-workflow-details/archived-workflow-details.tsx:250]: I think this is the logic you want:

https://github.com/argoproj/argo-workflows/blob/ab786ecba6eb3e9d3fa7a717ded42727b8b64df8/workflow/util/util.go#L629

**Review by alexec — CHANGES_REQUESTED** [review]
please update cloneLabelsSafely to match the go code, I expect you need to run `yarn lint` and `yarn build` on the UI too

**Comment by christopheblin** [comment]
@alexec do you expect the whole logic with memoized ? or "just" the part where we copy relevant labels ?  or also keep annotations and OwnerReference ?

note : I dont understand every bit in the go part, especially I am not sure "memoize" make sense at all in the archive use case...

**Comment by alexec** [comment]
> @alexec do you expect the whole logic with memoized ? or "just" the part where we copy relevant labels ? or also keep annotations and OwnerReference ?
> 
> note : I dont understand every bit in the go part, especially I am not sure "memoize" make sense at all in the archive use case...

I think just look at the parts about labels.

**Comment by sarabala1979** [comment]
@christopheblin When can we expect the PR to be addressed comments and ready for review?

**Comment by christopheblin** [comment]
Sorry, I switched to another subject and I currently have no time to move forward this PR :(

**Comment by tooptoop4** [comment]
where @christopheblin

**Comment by agilgur5** [comment]
> where @christopheblin

@tooptoop4 this page no longer exists after #11121 and resubmitting uses the resubmit endpoint these days which handles cloning labels after #3638. See also https://github.com/argoproj/argo-workflows/pull/11083

**Pull request closed without merging**